test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -0,0 +1,887 @@
+        self.env_hook_exclude = {}
+        self.penv = {}
+        self.read_env(job_outfile, "hook")
+
+        # Validate
+        self.common_validate()
+
+        # Verify the env variables in logs too
+        self.common_log_match("mom")
+
+    def test_execjob_pro(self):
+        """
+        Test that environment variable not gets truncated
+        for execjob_prologue hook
+        """
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.walltime': 10}
+        script = []
+        script += ['#PBS -V']
+        script += ['env\n']
+        script += ['sleep 10\n']
+
+        # Submit a job without hooks in the system
+        jid = self.create_and_submit_job(attribs=a, content=script)
+        qstat = self.server.status(JOB, ATTR_o, id=jid)
+        job_outfile = qstat[0][ATTR_o].split(':')[1]
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid, offset=10)
+
+        # read the output file for env without hook
+        self.env_nohook = {}
+        self.env_nohook_exclude = {}
+        self.penv = {}
+        self.read_env(job_outfile, "nohook")
+
+        # Now start the hooks
+        hook_name = "test_pro"
+        hook_body = """
+import pbs
+e = pbs.event()
+j = e.job
+j.Variable_List["happy"] = "days"
+pbs.logmsg(pbs.LOG_DEBUG,"Variable_List is %s" % (j.Variable_List,))
+"""
+        a2 = {'event': "execjob_prologue", 'enabled': "true", 'debug': "true"}
+
+        rv = self.server.create_import_hook(
+            hook_name,
+            a2,
+            hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+        # Submit a job with hooks in the system
+        jid2 = self.create_and_submit_job(attribs=a, content=script)
+        qstat = self.server.status(JOB, ATTR_o, id=jid2)
+        job_outfile = qstat[0][ATTR_o].split(':')[1]
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid2, offset=10)
+
+        # Read the job ouput file
+        self.env_hook = {}
+        self.env_hook_exclude = {}
+        self.penv = {}
+        self.read_env(job_outfile, "hook")
+
+        # Validate the env values with and without hook
+        self.common_validate()
+
+        # compare the values in mom_logs as well
+        self.common_log_match("mom")
+
+    @checkModule("pexpect")
+    def test_interactive(self):
+        """
+        Test that interactive jobs do not have truncated environment
+        variable list with execjob_launch hook
+        """
+
+        self.exclude_env += ['happy']
+
+        # submit an interactive job without hook
+        cmd = 'env > /tmp/job_out1'
+        a = {ATTR_inter: '', self.ATTR_V: None}
+
+        interactive_script = [('hostname', '.*'), (cmd, '.*')]
+        jid = self.create_and_submit_job(
+            attribs=a,
+            content_interactive=interactive_script,
+            preserve_env=True)
+        # Once all commands sent and matched, job exits
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid, offset=10)
```

Comment:
Same as above, you're testing a launch hook.  Why does the job have to finish?  Go through all the tests and reevaluate if the job has to finish.  You probably only have to wait for an epilogue hook.

Reply:
As I mentioned earlier, it's looking for the job output file or output from 'env > <file_name>' to read_env() settings. The job must have exited to get the results.